File: src/main/java/com/soapdogg/lispInterpreter/evaluator/AtomNodeEvaluator.java
package com.soapdogg.lispInterpreter.evaluator;

import com.soapdogg.lispInterpreter.datamodels.AtomNode;
import com.soapdogg.lispInterpreter.datamodels.Node;

import java.util.Map;

public class AtomNodeEvaluator {

    public Node evaluate(
        final AtomNode atomNode,
        final Map<String, ? extends Node> variableNameToValueMap
    ) {
        String value = atomNode.getValue();
        if (variableNameToValueMap.containsKey(value)) return variableNameToValueMap.get(value);
        return atomNode;
    }
}

File: src/main/java/com/soapdogg/lispInterpreter/evaluator/AtomNodeEvaluator.kt
package com.soapdogg.lispInterpreter.evaluator

import com.soapdogg.lispInterpreter.datamodels.AtomNode
import com.soapdogg.lispInterpreter.datamodels.NodeV2

class AtomNodeEvaluator {
    fun evaluate(
        atomNode: AtomNode,
        variableNameToValueMap: Map<String, NodeV2>
    ): NodeV2 {
        val value = atomNode.value
        return variableNameToValueMap.getOrDefault(value, atomNode)
    }
}